### ISSUE TITLE
Evaluation code

### ISSUE THREAD

**Issue by ChidanandKumarKS** [TimoStoff/event_cnn_minimal#20]
is evaluation code available with the mentioned pretrained model

**Comment by midofalasol** [comment]
Hello, I also encountered the same evaluation indicator calculation problem.

> [skimage.metrics.structural_similarity](https://scikit-image.org/docs/stable/api/skimage.metrics.html#skimage.metrics.structural_similarity)(im1, im2, *, win_size=None, gradient=False, data_range=None, channel_axis=None, multichannel=False, gaussian_weights=False, full=False, **kwargs)
> To match the implementation of Wang et. al. [38], set gaussian_weights to True, sigma to 1.5, and use_sample_covariance to False.

When I calculated the SSIM according to the above method, the result obtained was far from the data in the paper. Did you get a result similar to that in the paper?

**Comment by ChidanandKumarKS** [comment]
@midofalasol ,

I planned to use SPADE-E2VID evaluation code based on the link.
from skimage.measure import compare_mse, compare_ssim

https://github.com/RodrigoGantier/SPADE_E2VID/blob/817189ad2c9e84485ec5f395f9b3305fc2d02a52/benchmark.py#L6

Results are SPADE-E2VID are reproducible. Hence will be using their evaluation code. My GPUs are still training the model with the event_cnn_mimimal author's configuration. Do you know how long it will take since u trained it already? Iam using 1 GPU with 3090

**Comment by midofalasol** [comment]
> @midofalasol ,
> 
> I planned to use SPADE-E2VID evaluation code based on the link. from skimage.measure import compare_mse, compare_ssim
> 
> https://github.com/RodrigoGantier/SPADE_E2VID/blob/817189ad2c9e84485ec5f395f9b3305fc2d02a52/benchmark.py#L6
> 
> Results are SPADE-E2VID are reproducible. Hence will be using their evaluation code. My GPUs are still training the model with the event_cnn_mimimal author's configuration. Do you know how long it will take since u trained it already? Iam using 1 GPU with 3090

Stuck by my hardware environment, I haven't implemented it yet. I don’t know if the reproduced result of the evaluation code you used (pretraining checkpoint: reconstruction_model.pth) is consistent with the paper?

**Comment by ChidanandKumarKS** [comment]
I didn't use author's pretrained model so far since i want to check whether authors training code works properly by verifying losses. 
Once trained completely, i will cross verify the results of authors pretrained_model with the one i have trained. It is yet to do

**Comment by Adeelyousaf** [comment]
> is evaluation code available with the mentioned pretrained model

hi, I am training the event_cnn model, kindly if you could guide what to pass at data_file in reconstruction.json

**Comment by ercanburak** [comment]
For anyone seeking quantitative evaluation, you can utilize [EVREAL](https://github.com/ercanburak/EVREAL), our library designed to evaluate and analyze PyTorch-based event-based video reconstruction methods (including the methods from this repo, E2VID+ and FireNet+).
